src/commands/utils/flinkStatements.ts

```diff
@@ -0,0 +1,135 @@
+import {
+  CreateSqlv1Statement201Response,
+  CreateSqlv1StatementOperationRequest,
+  CreateSqlv1StatementRequest,
+  CreateSqlv1StatementRequestApiVersionEnum,
+  CreateSqlv1StatementRequestKindEnum,
+} from "../../clients/flinkSql";
+import { CCloudResourceLoader } from "../../loaders";
+import { CCloudFlinkComputePool } from "../../models/flinkComputePool";
+import { getSidecar } from "../../sidecar";
+
+export class FlinkSpecProperties {
+  currentCatalog: string | undefined = undefined;
+  currentDatabase: string | undefined = undefined;
+  localTimezone: string | undefined = undefined;
+
+  constructor(
+    options: Pick<
+      Partial<FlinkSpecProperties>,
+      "currentCatalog" | "currentDatabase" | "localTimezone"
+    >,
+  ) {
+    this.currentCatalog = options.currentCatalog;
+    this.currentDatabase = options.currentDatabase;
+    this.localTimezone = options.localTimezone;
+  }
+
+  static fromProperties(properties: Record<string, string>): FlinkSpecProperties {
+    const currentCatalog = properties["sql.current-catalog"];
+    const currentDatabase = properties["sql.current-database"];
+    const localTimezone = properties["sql.local-time-zone"];
+    return new FlinkSpecProperties({
+      currentCatalog,
+      currentDatabase,
+      localTimezone,
+    });
+  }
+
+  toProperties(): Record<string, string> {
+    const properties: Record<string, string> = {};
+    if (this.currentCatalog) {
+      properties["sql.current-catalog"] = this.currentCatalog;
+    }
+    if (this.currentDatabase) {
+      properties["sql.current-database"] = this.currentDatabase;
+    }
+    if (this.localTimezone) {
+      properties["sql.local-time-zone"] = this.localTimezone;
+    }
+    return properties;
+  }
+
+  /**
+   * Return new properties set based on union of this and provided other, preferring any value
+   * set in other over this.
+   */
+  union(other: FlinkSpecProperties): FlinkSpecProperties {
+    const merged = new FlinkSpecProperties({
+      currentCatalog: other.currentCatalog || this.currentCatalog,
+      currentDatabase: other.currentDatabase || this.currentDatabase,
+      localTimezone: other.localTimezone || this.localTimezone,
+    });
+    return merged;
+  }
+}
+
+/**
+ * Return a suitable (and unique) name for a Flink statement to submit.
+ *
+ * @param statement
+ */
+export async function determineFlinkStatementName(): Promise<string> {
```

Comment:
Minor: may be helpful to see an example of what this format ends up looking like in the docstring/comment